backend/atlas_backend/views.py

```diff
@@ -254,111 +326,272 @@ def start_challenge(request, challenge_id):
             {'error': 'You must be in a team to start challenges'},
             status=status.HTTP_400_BAD_REQUEST
         )
-        
+
     challenge = get_object_or_404(Challenge, id=challenge_id)
-    
+
     existing_container = Container.objects.filter(
         team=request.user.team,
         challenge=challenge,
-        status='running'
+        created_at__gte=datetime.now() - timedelta(minutes=10)
     ).first()
-    
+
     if existing_container:
         return Response({
             'host': existing_container.ssh_host,
             'port': existing_container.ssh_port,
-            'username': existing_container.ssh_user,
-            'container_id': existing_container.id,
-            'ssh_key': existing_container.ssh_key
+            'ssh_user': existing_container.ssh_user,
+            'ssh_password': existing_container.ssh_password,
+            'created_at': existing_container.created_at,
         })
-    
+
     try:
-        docker_mgr = DockerManager()
-        
-        container_info = docker_mgr.create_container(
+        client = DockerPlugin()
+
+        container_id, password = client.run_container(
             challenge.docker_image,
-            team_id=request.user.team.id,
-            challenge_id=challenge_id
+            port=22,
+            container_name=f"{request.user.team.name.replace(" ", "_")}-{challenge.title.replace(" ", "_")}"
         )
-        
+
+        while True:
+            ports = client.get_container_ports(container_id)
+            if ports:
+                break
+
```

Comment:
if no ports are returned this might become an issue
```suggestion
import time
        timeout = 30
        start_time = time.time()
        while True:
            ports = client.get_container_ports(container_id)
            if ports:
                break
            time.sleep(1)
            if time.time() - start_time > timeout:
                return Response(
                    {'error': 'Timeout waiting for container ports'},
                    status=status.HTTP_500_INTERNAL_SERVER_ERROR
                )
```